modal/_functions.py

```diff
@@ -1865,6 +1902,47 @@ async def get(self, timeout: Optional[float] = None, *, index: int = 0) -> Retur
         """
         return await self._invocation().poll_function(timeout=timeout, index=index)
 
+    @live_method_gen
+    async def iter(self, *args) -> AsyncIterator[Tuple[int, ReturnType]]:
+        """Iterate in-order over (index, result) pairs of the function call.
+        Optionally, specify a range [start, end) to iterate over.
```

Comment:
Nit; the one-line summary should be a single line and should have a blank before any additional explanation.